tribits/common_tpls/FindTPLNetcdf.cmake

```diff
@@ -37,105 +37,128 @@
 # ************************************************************************
 # @HEADER
 
-#
-# First, set up the variables for the (backward-compatible) TriBITS way of
-# finding Netcdf.  These are used in case find_package(NetCDF ...) is not
-# called or does not find NetCDF.  Also, these variables need to be non-null
-# in order to trigger the right behavior in the function
-# tribits_tpl_find_include_dirs_and_libraries().
-#
+cmake_policy(SET CMP0074 NEW)
 
-set(REQUIRED_HEADERS netcdf.h)
-set(REQUIRED_LIBS_NAMES netcdf)
+set(Netcdf_ALLOW_MODERN FALSE CACHE BOOL "Allow finding Netcdf as a modern CMake config file with exported targets (and only this way)")
 
-if (TPL_ENABLE_MPI)
-  set(REQUIRED_LIBS_NAMES ${REQUIRED_LIBS_NAMES} pnetcdf)
-endif()
+if (Netcdf_ALLOW_MODERN AND HDF5_FOUND_MODERN_CONFIG_FILE)
```

Comment:
After seeing how to use this in SEACAS -- 
This does not seem to work *if* HDF5 is added as a dependency only because of netCDF since at this point, I don't think that we have yet found HDF5, so `HDF5_FOUND_MODERN_CONFIG_FILE` will not be set?

At least for SEACAS, I think that the `Netcdf_ALLOW_MODERN` should drive this and determine whether to find HDF5 the modern/legacy way?  Does that make sense?

Or, do I reorder things such that CGNS (which depends on HDF5) comes first and then netCDF?
@bartlettroscoe @ibaned

Reply:
> Or, do I reorder things such that CGNS (which depends on HDF5) comes first and then netCDF?

@gsjaardema, you specify the dependencies as: HDF5 <-- CGNS <-- Netcdf

See:

* https://tribitspub.github.io/TriBITS/users_guide/index.html#tpldefsdir-findtpl-tplname-dependencies-cmake

It looks like GGNS already has a required dependency on HDF5 as per:

https://github.com/TriBITSPub/TriBITS/blob/9fd56d2afdf1c8e38da30afda1fbb7b7c393e28c/tribits/common_tpls/FindTPLCGNSDependencies.cmake#L1-L2

But I had no idea that Netcdf depended on CGNS?

Also not that, currently, TriBITS does **not** automatically enable upstream TPL dependencies for an enabled TPL.  So the configure script will need to manually enable all of the TPLs it wants.  (See NOTES at bottom of slide 16 "TriBITS External Packages/TPLs Dependencies (New!)" in the presentation [TriBITS Modernization Update](https://trilinos.github.io/pages/community/trilinos_user_meetings/TUG_2022/2022-10-27/01_TribitsModernization_TUG_2022-10-27.pdf) I gave last Thursday, Oct 27, 2022.)